hnn_core/network.py

```diff
@@ -299,6 +299,40 @@ def pick_connection(net, src_gids=None, target_gids=None,
     return sorted(conn_set)
 
 
+def _get_cell_index_by_synapse_type(net):
+    """Returns the indexes of excitatory and inhibitory cells in the network.
+
+    This function extracts the source GIDs (Global Identifiers) of excitatory
+    and inhibitory cells based on their connection types. Excitatory cells are
+    identified by their synaptic connections using AMPA and NMDA receptors,
+    while inhibitory cells are identified by their connections using GABAA and
+    GABAB receptors.
+
+    Parameters
+    ----------
+    net : Instance of Network object
+        The Network object
+
+    Returns
+    -------
+    tuple: A tuple containing two lists:
+        - e_cells (list): The source GIDs of excitatory cells.
+        - i_cells (list): The source GIDs of inhibitory cells.
+    """
+
+    def list_src_gids(indices):
+        return np.concatenate([list(net.connectivity[conn_idx]['src_gids'])
+                               for conn_idx in indices]).tolist()
+
+    e_conns = pick_connection(net, receptor=['ampa', 'nmda'])
```

Comment:
```suggestion
    picks_e = pick_connection(net, receptor=['ampa', 'nmda'])
```

I like to use `e_conn` after the indexing, i.e.:

```py
e_conns = [net.connectivity[p] for p in picks_e]
```

so it's clear from the variable name that one is an element of `net.connectivity` and the other is simply an index